src/utils/ObjectUtils.ts

```diff
@@ -90,24 +83,14 @@ function _equal(obj1: any, obj2: any, shallow: boolean = true): boolean {
       return false;
     }
     if (obj1.hasOwnProperty(property) && obj2.hasOwnProperty(property)) {
-      if (shallow) {
-        if (obj1[property] !== obj2[property]) {
-          return false;
-        }
-      } else {
-        if (!_equal(obj1[property], obj2[property], false)) {
+      if (isImmutable(obj1[property]) && isImmutable(obj2[property])) {
+        if (!obj1[property].equals(obj2[property])) {
           return false;
         }
+      } else if (obj1[property] !== obj2[property]) {
```

Comment:
Could drop the else and just `return obj1[property] === obj2[property]`

Reply:
unfortunately not, since this is in the middle of the loop where we iterate through the object's properties, we need to continue iteration in the `true` case or else objects' equality is determined based on the equality of the first property accessed